server/src/main/java/io/unitycatalog/server/persist/utils/UriUtils.java

```diff
@@ -40,17 +36,25 @@ private enum Operation {
   private UriUtils() {}
 
   private static String modelStorageRootCached;
+  private static String modelStorageRootPropertyCached;
 
-  public static void reset() {
+  private static void reset() {
+    modelStorageRootPropertyCached = null;
     modelStorageRootCached = null;
   }
 
   // Model specific storage root handlers and convenience methods
   private static String getModelStorageRoot() {
+    String currentModelStorageRoot = properties.getProperty("storage-root.models");
+    if (modelStorageRootPropertyCached != currentModelStorageRoot) {
+      // This means the property has been updated from the previous read, or this is the first time
+      // reading it
+      reset();
+    }
     if (modelStorageRootCached != null) {
```

Comment:
so we can't switch from project root directory to a storage root specified via properties ?